lib/rubocop/config.rb

```diff
@@ -172,8 +173,16 @@ def base_dir_for_path_parameters
 
     private
 
+    def validate_section_presence(name)
+      return unless @hash.key?(name) && @hash[name].nil?
+      fail ValidationError,
+           "empty section #{name} found " \
```

Comment:
It could probably be achieved where I initially added the code to silently remove the empty sections (ConfigLoader#load_yaml_configuration).

Not sure it's worth it, though, because retrieving the line number with psych (the yaml implementation under YAML) seems quite a pain in the ass.

It imply:
* writing a handler that keep track of every scalar and the following mark
* check that the initial mapping contained in the psych document contains only not nil values
* raise the error if needed, with the line number given by the mark following the faulty node

Not sure I wanna be in for this one :)